webpack.config.js

```diff
@@ -378,19 +403,19 @@ const postEditorBlock = {
   },
 };
 
-module.exports = [adminConfig, publicConfig, migratorConfig, formPreviewConfig, testConfig, postEditorBlock].map((config) => {
+module.exports = [publicConfig, adminConfig, migratorConfig, formPreviewConfig, testConfig, postEditorBlock].map((config, index) => {
   if (config.name !== 'test') {
     config.plugins = config.plugins || [];
     config.plugins.push(
-      new webpackManifestPlugin({
+      new WebpackManifestPlugin({
         // create single manifest file for all Webpack configs
         seed: manifestSeed,
       })
     );
   }
-  // Clean output paths before build
-  if (config.output && config.output.path) {
-    del.sync([path.resolve(config.output.path, '**/*')]);
+  // Cleanup output path but only once to keep output files for subsequent builds
+  if (index === 0) {
+    config.plugins.unshift(new CleanWebpackPlugin());
```

Comment:
Could we use https://webpack.js.org/configuration/output/#outputclean or are we on a use case that still needs   clean-webpack-plugin https://github.com/johnagan/clean-webpack-plugin/issues/197?

Reply:
I didn't know this option. Thanks! It will do the job!

Reply:
Hmm. After some testing, I found out that in free we probably can't use the clean-webpack-plugin or the output.clean option.

We need to share the same output path (assets/build/js) in more configs in order to generate a common manifest.json file. The cleanup plugin and also built-in output.clean setting does the cleanup on emitting and that causes that they delete files we want to keep even though I configured cleanup only for the first config. 

Using different output paths solves the problem with cleanup but it causes that we generate a bunch of manifest.json files instead of one.

As a simple workaround, I chose cleaning the output directory before we trigger the Webpack build in the Robofile.

Reply:
Sounds good to me! Thanks for testing the other option and for all the details